src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -166,7 +166,8 @@ void CodeGen::genEmitGSCookieCheck(bool pushReg)
             // returned in implicit RetBuf. If we reached here, we should not have
             // a RetBuf and the return type should not be a struct.
             assert(compiler->info.compRetBuffArg == BAD_VAR_NUM);
-            assert(!varTypeIsStruct(compiler->info.compRetNativeType));
+            // __m128 should be returned in XMM0
```

Comment:
The comment above should be changed, rather than simply adding on a comment about `__m128`. Also, that is the name of the native type, and I think it's clearer to use the JIT name for the type in the comments (i.e. `TYP_SIMD16`) for consistency.